### PR TITLE
Change how PRs are skipped so that required checks won't get stuck in pending

### PR DESCRIPTION
**Pull request by idodod**
The current configuration might causes jobs that are marked as required checks to get stuck in pending.
This PR aims to fix it.

See the readme added in this PR for more information.

Also fixing:
* workflows not running properly on push to main
* python tutorial tests

### CODE REVIEW

**Review by mikejholly — APPROVED** [review]
Sounds reasonable.

But, I'm wondering if improving Earthly & test performance (caching, etc.) are the "correct" areas to improve. Maybe a bigger conversation for later.

**Comment by idodod** [comment]
> Sounds reasonable.
> 
> But, I'm wondering if improving Earthly & test performance (caching, etc.) are the "correct" areas to improve. Maybe a bigger conversation for later.

Yeah, I would think that we would want to do both. This doesn't "hide" a slowness problem as most PRs will still take time, it should just allow us to merge PRs that only have to do with documentation or unrelated configuration faster.

File: .github/workflows/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+So, in order to work around the above-mentioned issue, we need to avoid
+skipping the parent job. Instead, we evaluate the boolean as before, and pass it as  
+an input argument to the child job. As a matter of standardization, we call this input arg `CONTINUE`.
+This argument is then used to conditionally run the _child_ job instead of the _parent_ job,  
+and so if the child job is skipped, even if it is marked as a required check, it won't get stuck in `pending`.
+
+Lastly, because most jobs are dependent on (`needs` directive) `build-earthly` job, if the latter is skipped, by default - so will the dependent jobs.  
+As described before, that is something we need to avoid. To change that default configuration, we change
+the default condition of each dependent job to `$ {{ !failure() }}` which will allow the dependent jobs to run whether the dependency job was successful or if it was skipped.
+
```
Comment (alexcb) [inline .github/workflows/README.md:47]: nit: remove trailing newlines (while still remaining posix compatible)

File: .github/workflows/reusable-wait-block-override.yml [hunk 2]
```diff
@@ -15,10 +15,14 @@ on:
       RUNS_ON:
         required: true
         type: string
-
+      CONTINUE:
```
Comment (alexcb) [inline .github/workflows/reusable-wait-block-override.yml:18]: can we pick a different variable name for this? `CONTINUE` makes me think we're inside a loop and will be re-running the loop.

what if instead we called this `PERFORM_TEST` instead? Or alternatively `SKIP_TEST`, which would then require us to also adjust the values we pass in.

**Review by alexcb — COMMENTED** [review]
nice work! just one suggestion.

File: .github/workflows/reusable-wait-block-override.yml (hunk 2 above)
Comment (idodod) [reply .github/workflows/reusable-wait-block-override.yml:18]: will do!

**Review by alexcb — APPROVED** [review]
awesome!

**Pull request merged**